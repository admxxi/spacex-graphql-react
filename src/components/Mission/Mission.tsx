import React from 'react'
import styled, { withTheme } from 'styled-components'
import { Launch as ILaunch, Mission as IMission } from '../../@types/graphql'
import { mySlugify } from '../../helpers/index'
import Link from 'next/link'
interface LaunchProps {
  mission: ILaunch
}

const Style = styled.div`
  cursor: pointer;

  .media-content > p {
    margin-top: 0;
  }
`

const Mission = ({ mission }: LaunchProps): JSX.Element => {
  const launchDateLocale = new Date(
    mission.launch_date_utc
  ).toLocaleDateString()

  const launchDetails =
    mission.details && mission.details.length > 120
      ? mission.details.substring(0, 117) + '...'
      : mission.details
  return (
    <Style>
      <Link
        href={{
          pathname: '/mission-detail/[slug]',
          query: {
            slug: `${mySlugify(String(mission.id), true)}`
          }
        }}
      >
        <article className="card">
          <div className="card-content">
            <div className="media is-align-content-center">
              <div className="media-content">
                <a className="is-link is-3">{mission.mission_name}</a>
                <p className="subtitle is-4 mt-2">
                  {launchDetails ? launchDetails : 'No description available'}
                </p>
                <span className="ma-1 subtitle is-5 is-italic has-text-right mx-1">
                  {launchDateLocale}
                </span>
              </div>
            </div>
          </div>
        </article>
      </Link>
    </Style>
  )
}

export default Mission
