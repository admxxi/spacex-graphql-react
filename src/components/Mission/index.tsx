import React from 'react'
import { Launch as ILaunch } from '../../@types/graphql'

interface MissionProps {
  launch: ILaunch
}

const Mission = ({ launch }: MissionProps) => {
  console.log(launch)
  const flickerImages = launch.links?.flickr_images
    ? launch.links?.flickr_images
    : []

  return (
    <div className="card d-inline-block">
      {flickerImages.length && (
        <div className="card-image">
          <figure className="image is-4by3">
            <img
              src={flickerImages[0]}
              alt="Placeholder image"
              width="100%"
              height="auto"
            />
          </figure>
        </div>
      )}
      <div className="card-content">
        {launch.links?.mission_patch && (
          <div className="media">
            <div className="media-left">
              <figure className="image is-48x48">
                <img
                  src={launch.links?.mission_patch}
                  alt="Placeholder image"
                  width="100%"
                  height="auto"
                />
              </figure>
            </div>

            <div className="media-content">
              <p className="title is-4">Patch</p>
            </div>
          </div>
        )}
        <div className="content">
          {launch.id}
          {launch.details}
          {launch.links?.video_link && (
            <a href={launch.links?.video_link}>Youtube video</a>
          )}
        </div>
      </div>
    </div>
  )
}

export default Mission
