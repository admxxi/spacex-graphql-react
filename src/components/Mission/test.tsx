import React from 'react'
import renderer from 'react-test-renderer'
import Launch from './Launch'

it('renders correctly', () => {
  const tree = renderer.create(<Launch />).toJSON()
  expect(tree).toMatchSnapshot()
})
