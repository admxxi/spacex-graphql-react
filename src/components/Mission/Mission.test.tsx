import React from 'react';
import renderer from 'react-test-renderer';
import Mission from './index';

it('renders correctly', () => {
  const tree = renderer.create(<Mission />).toJSON()
  expect(tree).toMatchSnapshot()
});