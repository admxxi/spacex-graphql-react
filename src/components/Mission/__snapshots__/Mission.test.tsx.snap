// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <figure
      className="image is-4by3"
    >
      <img
        alt="Placeholder image"
        src="https://bulma.io/images/placeholders/1280x960.png"
      />
    </figure>
  </div>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            alt="Placeholder image"
            src="https://bulma.io/images/placeholders/96x96.png"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <p
          className="title is-4"
        >
          John Smith
        </p>
        <p
          className="subtitle is-6"
        >
          @johnsmith
        </p>
      </div>
    </div>
    <div
      className="content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. 
      <a>
        @bulmaio
      </a>
      .
      <a
        href="#"
      >
        #css
      </a>
       
      <a
        href="#"
      >
        #responsive
      </a>
    </div>
  </div>
</div>
`;
